libs/constrained-generators/src/Constrained/Spec/Map.hs

```diff
@@ -334,7 +336,14 @@ instance BaseUniverse fn => Functions (MapFn fn) fn where
           | HOLE :? Value (m :: Map k v) :> Nil <- ctx ->
               if Nothing `conformsToSpec` spec
                 then notMemberSpec [k | (k, v) <- Map.toList m, not $ Just v `conformsToSpec` spec]
-                else MemberSpec $ nubOrd $ Map.keys $ Map.filter (`conformsToSpec` spec) (Just <$> m)
+                else
+                  memberSpecList
+                    (nubOrd $ Map.keys $ Map.filter (`conformsToSpec` spec) (Just <$> m))
```

Comment:
Redundant `nubOrd` and `<$>`. `Map.keys` will already return a list with no duplicates and `<$>` can be avoided too:
```suggestion
                    (Map.keys $ Map.filter ((`conformsToSpec` spec) . Just) m)
```